Problem in `_Quaxify.__call__`

`_Quaxify.__call__` is converting things to tracers when they shouldn't be tracers. For example, here's a stable MWE

```python
import jax.numpy as jnp
import quax

xbool = jnp.array([ True, False,  True], dtype=bool)
x1 = jnp.array([1., 2., 3.], dtype=float)

compress = quax.quaxify(jnp.compress)
compress(xbool, x1) 

```

Here's the traceback. I think this is saying that some stuff are becoming tracer, and that's triggering the error. 

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../quax/quax/_core.py:311: in __call__
    out = fn(*args, **kwargs)
        args       = (Array([ True, False,  True], dtype=bool), Array([1., 2., 3.], dtype=float32))
        dynamic    = (<function compress at 0x1084b2340>, (Array([ True, False,  True], dtype=bool), Array([1., 2., 3.], dtype=float32)), {})
        fn         = <function compress at 0x1084b2340>
        kwargs     = {}
        parent_trace = EvalTrace
        self       = _Quaxify(fn=<function compress>, filter_spec=True, dynamic=False)
        static     = (None, (None, None), {})
        tag        = <jax._src.core.TraceTag object at 0x110d3be90>
        trace      = _QuaxTrace
.venv/lib/python3.11/site-packages/jax/_src/numpy/lax_numpy.py:9595: in compress
    if reductions.any(extra):
        a          = Array([1., 2., 3.], dtype=float32)
        arr        = Traced<ShapedArray(float32[3])>with<_QuaxTrace> with
  value = _DenseArrayValue(array=f32[3])
        axis       = 0
        condition  = Array([ True, False,  True], dtype=bool)
        condition_arr = Array([ True, False,  True], dtype=bool)
        extra      = Traced<ShapedArray(bool[0])>with<_QuaxTrace> with
  value = _DenseArrayValue(array=bool[0])
        fill_value = Traced<ShapedArray(int32[], weak_type=True)>with<_QuaxTrace> with
  value = _DenseArrayValue(array=weak_i32[])
        out        = None
        size       = None
.venv/lib/python3.11/site-packages/jax/_src/core.py:842: in __bool__
    return self.aval._bool(self)
        self       = Traced<ShapedArray(bool[])>with<_QuaxTrace> with
  value = _DenseArrayValue(array=bool[])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ShapedArray(bool[]), arg = Traced<ShapedArray(bool[])>with<_QuaxTrace> with
  value = _DenseArrayValue(array=bool[])

    def error(self, arg):
>     raise TracerBoolConversionError(arg)
E     jax.errors.TracerBoolConversionError: Attempted boolean conversion of traced array with shape bool[].
E     See https://jax.readthedocs.io/en/latest/errors.html#jax.errors.TracerBoolConversionError

arg        = Traced<ShapedArray(bool[])>with<_QuaxTrace> with
  value = _DenseArrayValue(array=bool[])
self       = ShapedArray(bool[])
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -25,7 +25,7 @@ dependencies = [
   "optype>=0.8.0",
   'plum-dispatch>=2.5.7; python_version < "3.14"',
   'plum-dispatch>=2.9.0; python_version >= "3.14"',
-  "quax>=0.4.0",
+  "quax>=0.4.1",
 ]
 description = "Pre-quaxed libraries for multiple dispatch over abstract array types in JAX"
 dynamic = ["version"]
@@ -45,7 +45,7 @@ quaxed-make-stubs = "quaxed._tools.make_stubs:main"
 
 [build-system]
 build-backend = "hatchling.build"
-requires      = ["hatchling", "hatch-vcs", "jax>=0.7.2", "quax>=0.4.0"]
+requires      = ["hatchling", "hatch-vcs", "jax>=0.7.2", "quax>=0.4.1"]
 
 
 [dependency-groups]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -26,7 +26,8 @@ dependencies = [
   "optype>=0.8.0",
   'plum-dispatch>=2.5.7; python_version < "3.14"',
   'plum-dispatch>=2.9.0; python_version >= "3.14"',
-  "quax>=0.3.2",
+  'quax>=0.3.2; python_version < "3.14"',
+  'quax>=0.4.0; python_version >= "3.14"',
 ]
 description = "Pre-quaxed libraries for multiple dispatch over abstract array types in JAX"
 dynamic = ["version"]
@@ -51,7 +52,8 @@ requires = [
   "hatch-vcs",
   'jax>=0.5.3; python_version < "3.14"',
   'jax>=0.7.2; python_version >= "3.14"',
-  "quax>=0.3.2",
+  'quax>=0.3.2; python_version < "3.14"',
+  'quax>=0.4.0; python_version >= "3.14"',
 ]
 
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -20,14 +20,12 @@ classifiers = [
 ]
 dependencies = [
   "equinox>=0.13.2",
-  'jax>=0.5.3; python_version < "3.14"',
-  'jax>=0.7.2; python_version >= "3.14"',
-  "jaxtyping>=0.3.1",
+  "jax>=0.7.2",
+  "jaxtyping>=0.3.3",
   "optype>=0.8.0",
   'plum-dispatch>=2.5.7; python_version < "3.14"',
   'plum-dispatch>=2.9.0; python_version >= "3.14"',
-  'quax>=0.3.2; python_version < "3.14"',
-  'quax>=0.4.0; python_version >= "3.14"',
+  "quax>=0.4.0",
 ]
 description = "Pre-quaxed libraries for multiple dispatch over abstract array types in JAX"
 dynamic = ["version"]
@@ -47,14 +45,7 @@ quaxed-make-stubs = "quaxed._tools.make_stubs:main"
 
 [build-system]
 build-backend = "hatchling.build"
-requires = [
-  "hatchling",
-  "hatch-vcs",
-  'jax>=0.5.3; python_version < "3.14"',
-  'jax>=0.7.2; python_version >= "3.14"',
-  'quax>=0.3.2; python_version < "3.14"',
-  'quax>=0.4.0; python_version >= "3.14"',
-]
+requires      = ["hatchling", "hatch-vcs", "jax>=0.7.2", "quax>=0.4.0"]
 
 
 [dependency-groups]

--- a/tests/unit/test_numpy/test_jax.py
+++ b/tests/unit/test_numpy/test_jax.py
@@ -15,9 +15,13 @@ from quax._compat import JAX_VERSION
 import quaxed.numpy as qnp
 
 NUMPY_VERSION = Version(np.__version__)
+QUAX_VERSION = Version(quax.__version__)
 
+# quax v0.4.0 fixed https://github.com/patrick-kidger/quax/issues/58 for the
+# JAX-input path, so these functions now pass. Keep the xfail for older quax.
 xfail_quax58 = pytest.mark.xfail(
-    reason="https://github.com/patrick-kidger/quax/issues/58"
+    condition=Version("0.4") > QUAX_VERSION,
+    reason="https://github.com/patrick-kidger/quax/issues/58",
 )
 mark_todo = pytest.mark.skip("TODO")
 xfail_numpy_2_3_implicit_conversion = pytest.mark.xfail(

--- a/tests/unit/test_numpy/test_jax.py
+++ b/tests/unit/test_numpy/test_jax.py
@@ -15,14 +15,7 @@ from quax._compat import JAX_VERSION
 import quaxed.numpy as qnp
 
 NUMPY_VERSION = Version(np.__version__)
-QUAX_VERSION = Version(quax.__version__)
 
-# quax v0.4.0 fixed https://github.com/patrick-kidger/quax/issues/58 for the
-# JAX-input path, so these functions now pass. Keep the xfail for older quax.
-xfail_quax58 = pytest.mark.xfail(
-    condition=Version("0.4") > QUAX_VERSION,
-    reason="https://github.com/patrick-kidger/quax/issues/58",
-)
 mark_todo = pytest.mark.skip("TODO")
 xfail_numpy_2_3_implicit_conversion = pytest.mark.xfail(
     Version("2.3") <= NUMPY_VERSION,
@@ -93,7 +86,7 @@ xbool = jnp.array([True, False, True], dtype=bool)
         ("argpartition", (x, 1), {}),
         ("argsort", (x,), {}),
         *(
-            pytest.param("argwhere", (x,), {}, marks=xfail_quax58),
+            pytest.param("argwhere", (x,), {}),
             ("argwhere", (x,), {"size": x.size}),  # TODO: not need static size
         ),
         ("around", (x,), {}),
@@ -114,9 +107,7 @@ xbool = jnp.array([True, False, True], dtype=bool)
         ("average", (x,), {}),
         ("bartlett", (3,), {}),
         *(
-            pytest.param(
-                "bincount", (jnp.asarray([0, 1, 1, 2, 2, 2]),), {}, marks=xfail_quax58
-            ),
+            ("bincount", (jnp.asarray([0, 1, 1, 2, 2, 2]),), {}),
             ("bincount", (jnp.asarray([0, 1, 1, 2, 2, 2]),), {"length": 3}),
         ),
         ("bitwise_and", (xbool, xbool), {}),
@@ -139,14 +130,14 @@ xbool = jnp.array([True, False, True], dtype=bool)
         ),
         ("cbrt", (x,), {}),
         ("ceil", (x,), {}),
-        pytest.param("choose", (0, [x, x]), {}, marks=xfail_quax58),
+        pytest.param("choose", (0, [x, x]), {}),
         ("clip", (x, 1, 2), {}),
         ("column_stack", ([x, x],), {}),
         pytest.param("complex128", (1,), {}, marks=pytest.mark.xfail),
         ("complex64", (1,), {}),
         pytest.param("complex_", (1,), {}, marks=pytest.mark.xfail),
         pytest.param("complexfloating", (1,), {}, marks=pytest.mark.xfail),
-        pytest.param("compress", (xbool, x), {}, marks=xfail_quax58),
+        pytest.param("compress", (xbool, x), {}),
         ("concat", ([x, x],), {}),
         ("concatenate", ([x, x],), {}),
         ("conj", (x,), {}),
@@ -190,7 +181,7 @@ xbool = jnp.array([True, False, True], dtype=bool)
         ("exp2", (x,), {}),
         ("expand_dims", (x, 0), {}),
         ("expm1", (x,), {}),
-        pytest.param("extract", (jnp.array([True]), x), {}, marks=xfail_quax58),
+        pytest.param("extract", (jnp.array([True]), x), {}),
         ("eye", (2,), {}),
         ("fabs", (x,), {}),
         ("fill_diagonal", (jnp.eye(3), 2), {"inplace": False}),
@@ -201,7 +192,7 @@ xbool = jnp.array([True, False, True], dtype=bool)
             marks=[*xfail_deprecated_jax_0_9_0, skip_removed_jax_0_10_0],
         ),
         *(
-            pytest.param("flatnonzero", (x,), {}, marks=xfail_quax58),
+            pytest.param("flatnonzero", (x,), {}),
             ("flatnonzero", (x,), {"size": x.size}),
         ),
         ("flip", (x,), {}),
@@ -256,7 +247,7 @@ xbool = jnp.array([True, False, True], dtype=bool)
         ("insert", (x, 1, 2.0), {}),
         ("interp", (x[:, 0], x[:, 0], 2 * x[:, 0]), {}),
         *(
-            pytest.param("intersect1d", (x[:, 0], x[:, 0]), {}, marks=xfail_quax58),
+            pytest.param("intersect1d", (x[:, 0], x[:, 0]), {}),
             ("intersect1d", (x[:, 0], x[:, 0]), {"size": x[:, 0].size}),
         ),
         ("invert", (x.astype(int),), {}),
@@ -333,7 +324,7 @@ xbool = jnp.array([True, False, True], dtype=bool)
         ("negative", (x,), {}),
         ("nextafter", (x, y), {}),
         *(
-            pytest.param("nonzero", (x,), {}, marks=xfail_quax58),
+            pytest.param("nonzero", (x,), {}),
             ("nonzero", (x,), {"size": x.size}),
         ),
         ("not_equal", (x, y), {}),
@@ -366,12 +357,7 @@ xbool = jnp.array([True, False, True], dtype=bool)
         ("rad2deg", (x,), {}),
         ("radians", (x,), {}),
         ("ravel", (x,), {}),
-        pytest.param(
-            "ravel_multi_index",
-            (jnp.array([[0, 1], [0, 1]]), (2, 2)),
-            {},
-            marks=xfail_quax58,
-        ),
+        ("ravel_multi_index", (jnp.array([[0, 1], [0, 1]]), (2, 2)), {}),
         ("real", (x,), {}),
         ("reciprocal", (x,), {}),
         ("remainder", (x, y), {}),
@@ -383,7 +369,7 @@ xbool = jnp.array([True, False, True], dtype=bool)
         ("rint", (x,), {}),
         ("roll", (x, 4), {}),
         ("rollaxis", (x, -1), {}),
-        pytest.param("roots", (x[:, 0],), {}, marks=xfail_quax58),
+        pytest.param("roots", (x[:, 0],), {}),
         ("rot90", (x,), {}),
         ("round", (x,), {}),
         # pytest.param("round_", (x,), {}, marks=pytest.mark.deprecated),
@@ -392,11 +378,11 @@ xbool = jnp.array([True, False, True], dtype=bool)
         pytest.param("searchsorted", (x,), {}, marks=mark_todo),
         pytest.param("select", (x,), {}, marks=mark_todo),
         *(
-            pytest.param("setdiff1d", (x[:, 0], y[:, 0]), {}, marks=xfail_quax58),
+            pytest.param("setdiff1d", (x[:, 0], y[:, 0]), {}),
             ("setdiff1d", (x[:, 0], y[:, 0]), {"size": x[:, 0].size}),
         ),
         *(
-            pytest.param("setxor1d", (x[:, 0], y[:, 0]), {}, marks=xfail_quax58),
+            pytest.param("setxor1d", (x[:, 0], y[:, 0]), {}),
             ("setxor1d", (x[:, 0], y[:, 0]), {"size": x[:, 0].size}),
         ),
         ("shape", (x,), {}),
@@ -427,7 +413,7 @@ xbool = jnp.array([True, False, True], dtype=bool)
         ("transpose", (x,), {}),
         ("tril", (jnp.eye(4),), {}),
         ("tril_indices_from", (jnp.eye(4),), {}),
-        pytest.param(
+        (
             "trim_zeros",
             (
                 jnp.concatenate(
@@ -435,7 +421,6 @@ xbool = jnp.array([True, False, True], dtype=bool)
                 ),
             ),
             {},
-            marks=xfail_quax58,
         ),
         ("triu", (x,), {}),
         ("triu_indices_from", (x,), {}),
@@ -443,27 +428,27 @@ xbool = jnp.array([True, False, True], dtype=bool)
         ("trunc", (x,), {}),
         pytest.param("ufunc", (x,), {}, marks=mark_todo),
         *(
-            pytest.param("union1d", (x[:, 0], y[:, 0]), {}, marks=xfail_quax58),
+            pytest.param("union1d", (x[:, 0], y[:, 0]), {}),
             ("union1d", (x[:, 0], y[:, 0]), {"size": x[:, 0].size}),
         ),
         *(
-            pytest.param("unique", (x,), {}, marks=xfail_quax58),
+            pytest.param("unique", (x,), {}),
             ("unique", (x,), {"size": x.size}),
         ),
         *(
-            pytest.param("unique_all", (x,), {}, marks=xfail_quax58),
+            pytest.param("unique_all", (x,), {}),
             ("unique_all", (x,), {"size": x.size}),
         ),
         *(
-            pytest.param("unique_counts", (x,), {}, marks=xfail_quax58),
+            pytest.param("unique_counts", (x,), {}),
             ("unique_counts", (x,), {"size": x.size}),
         ),
         *(
-            pytest.param("unique_inverse", (x,), {}, marks=xfail_quax58),
+            pytest.param("unique_inverse", (x,), {}),
             ("unique_inverse", (x,), {"size": x.size}),
         ),
         *(
-            pytest.param("unique_values", (x,), {}, marks=xfail_quax58),
+            pytest.param("unique_values", (x,), {}),
             ("unique_values", (x,), {"size": x.size}),
         ),
         pytest.param(

